lib/core/diagnostics.js

```diff
@@ -23,7 +23,13 @@ const channels = {
   close: diagnosticsChannel.channel('undici:websocket:close'),
   socketError: diagnosticsChannel.channel('undici:websocket:socket_error'),
   ping: diagnosticsChannel.channel('undici:websocket:ping'),
-  pong: diagnosticsChannel.channel('undici:websocket:pong')
+  pong: diagnosticsChannel.channel('undici:websocket:pong'),
+  // Fetch channels
+  start: diagnosticsChannel.channel('undici:fetch:start'),
```

Comment:
We can do this in another PR, but maybe `debug` should also scope those events for the fetch flag

Reply:
@metcoder95  I can add it to this PR, can I add the logs to this if block `if (undiciDebugLog.enabled || fetchDebuglog.enabled)` and I can follow the log pattern as well?

Reply:
@metcoder95 I added the debug logs following the patterns that were already present, please let me know if they are okay